content/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Configuring OpenID Connect in PyPI
+shortTitle: OpenID Connect in PyPI
+intro: Use OpenID Connect within your workflows to authenticate with PyPI
+versions:
+  fpt: '*'
+  ghec: '*'
+  ghes: '*'
+type: tutorial
+topics:
+  - Security
+  - Actions
+---
+
+{% data reusables.actions.enterprise-beta %}
+{% data reusables.actions.enterprise-github-hosted-runners %}
+
+## Overview
+
+OpenID Connect (OIDC) allows your {% data variables.product.prodname_actions %} workflows to authenticate with [PyPI](https://pypi.org) to publish Python packages.
+
+This guide gives an overview of how to configure PyPI to trust {% data variables.product.prodname_dotcom %}'s OIDC as a federated identity, and demonstrates how to use this configuration in the [`pypa/gh-action-pypa-publish`](https://github.com/pypa/gh-action-pypa-publish) action to publish packages to PyPI (or other Python package indices) without any manual API token management.
+
+{% ifversion ghes %}
+{% note %}
+
+**Note:** PyPI currently only supports OIDC federation with GitHub's own hosted OIDC IdP, which is identified by its issuer (`https://token.actions.githubusercontent.com`). Self-hosted GitHub Enterprise instances are not supported.
+
+{% endnote %}
+{% endif %}
```

Comment:
Your understanding is correct! This functionality only works with GitHub's own first-party OIDC IdP for GitHub Actions, meaning that GHES instances won't be able to produce tokens that PyPI will accept.

I wasn't exactly sure how to filter on features like that, so if you could give me some pointers I'd be happy to make that change!

Reply:
@jc-clark there's another possible problem with using the action in such environment — many orgs would only allow "verified" actions and we have been unable to make it work. See https://github.com/pypa/gh-action-pypi-publish/issues/94 — maybe you could help get to the bottom of it?

Reply:
Thanks for the link @webknjaz! And for the confirmation @woodruffw. I can take another look at this PR today. 

Noting for myself that I wasn't right with my hunch for the broken link yesterday. But I do think I have an idea where it's coming from and will try another solution.

Reply:
👋 @woodruffw I removed the GHES versioning in the latest commit since GHES versions do not need to see this article. That simplifies things a bit! 

I think the link checker might be expecting to see this new article listed under the "[Security hardening your deployments](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments)" map topic. 

Try adding the `/configuring-openid-connect-in-pypi` relative path to the `children` list in [this index.md file](https://github.com/github/docs/blob/87ed0f7ed7cad3a9e628ce90182d33d891c245c4/content/actions/deployment/security-hardening-your-deployments/index.md?plain=1).